src/transform/CollapseTable.js

```diff
@@ -13,48 +13,25 @@ const SECTION_TOGGLED_EVENT_TYPE = 'section-toggled'
  * @return {!Array<string>}
  */
 const getTableHeader = (element, pageTitle) => {
```

Comment:
This method was ported pretty closely from the original Android implementation (I don't know who wrote it) and there's obviously FUNKINESS about it. You may want to ask one of the Android devs to verify functionality before merging or add a CollapseTransform demo if they don't have time.

Reply:
Ya I pinged Dmitry and Michael and Dmitry didn't recall (been a long time hehe) and I think Michael was out. 

It took a while to dissect but I'm pretty confident I grok what it was doing and that my replacement should produce same effect. Been testing it on various pages and so far no divergence... fingers crossed ;)